### PR TITLE
[Jupyter] PPS Metadata Format + Project Support

### PR DESCRIPTION
**Pull request by bbonenfant**
Format:
```
version: (str) version ID of the metadata format
config:
  image: (str) docker image (with tag)
  pipeline:
    name: (str) name of the pipeline
    project:
      name: (str) name of the pipeline's project
  requirements: (str) relative path to the requirements.txt file
  input_spec: (dict) JSON of a valid input spec
}
```

See jupyter-extension/jupyterlab_pachyderm/tests/data/TestNotebook.ipynb for example

Tasks:
- [x] Remove all SAME references
- [x] Update METADATA_KEY
- [x] Update BE to read new metadata format
- [x] Update FE to read/write new metadata format
- [x] Breakout input spec parsing into helper method
- [x] Add project support 

Future Tasks:
- [ ] Validate input spec
- [ ] Properly define an InputSpec type

Closes: https://pachyderm.atlassian.net/browse/INT-888 and https://pachyderm.atlassian.net/browse/INT-893 also potentially closes https://pachyderm.atlassian.net/browse/INT-908

### CODE REVIEW

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx [hunk 1]
```diff
@@ -22,24 +30,43 @@ export type usePipelineResponse = {
...
 ): usePipelineResponse => {
   const [loading, setLoading] = useState(false);
-  const [pipelineName, setPipelineName] = useState('');
+  const [pipeline, _setPipeline] = useState({name: ''} as Pipeline);
   const [imageName, setImageName] = useState('');
   const [inputSpec, setInputSpec] = useState('');
   const [requirements, setRequirements] = useState('');
   const [errorMessage, setErrorMessage] = useState('');
 
+  const setPipeline = (input: string) => {
```
Comment (bbonenfant) [inline jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:42]: I'd like to test this method individually, but I can't figure out how to do so. Any advice?

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx [hunk 2]
```diff
@@ -30,28 +34,46 @@ export type usePipelineResponse = {
...
   const [errorMessage, setErrorMessage] = useState('');
   const [responseMessage, setResponseMessage] = useState('');
   const [currentNotebook, setCurrentNotebook] = useState('None');
 
+  const setPipeline = (input: string) => {
+    if (input === '') {
+      _setPipeline({name: ''} as Pipeline);
+      return;
+    }
+    const parts = input.split(/\/(.*)/s, 2);
```
Comment (chainlink) [inline jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:53]: I assume yes, but is `/` a restricted character in pipeline names?

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx [hunk 3]
```diff
@@ -76,7 +98,7 @@ export const usePipeline = (
         }
       } catch (e) {
         if (e instanceof ServerConnection.ResponseError) {
-          setErrorMessage(e.message);
+          setErrorMessage('Error creating pipeline');
```
Comment (chainlink) [inline jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:101]: Don't we want to return the server message here? (if we're passing a nicer error from the backend)

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx [hunk 4]
```diff
@@ -5,15 +5,19 @@ import {ServerConnection} from '@jupyterlab/services';
...
   MountSettings,
+  Pipeline,
   PpsContext,
-  SameMetadata,
+  PpsMetadata,
 } from '../../../types';
 import {requestAPI} from '../../../../../handler';
+import {ReadonlyJSONObject} from '@lumino/coreutils';
+
+export const PPS_VERSION = 'v1.0.0';
```
Comment (chainlink) [inline jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:15]: We might want to find a better spot for this

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx [hunk 5]
```diff
@@ -30,28 +34,46 @@ export type usePipelineResponse = {
...
-    setPipelineName(ppsContext?.config?.metadata.name ?? '');
-    setRequirements(ppsContext?.config?.notebook.requirements ?? '');
+    setRequirements(ppsContext?.metadata?.config.requirements ?? '');
     setResponseMessage('');
-    if (ppsContext?.config?.run.input) {
-      const input = JSON.parse(ppsContext.config.run.input); //TODO: Catch errors
-      setInputSpec(YAML.stringify(input));
+    if (ppsContext?.metadata?.config.input_spec) {
+      try {
+        setInputSpec(YAML.stringify(ppsContext.metadata.config.input_spec));
```
Comment (chainlink) [inline jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:72]: Does this yaml stringify work with an empty input spec?

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx (hunk 2 above)
Comment (bbonenfant) [reply jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:53]: yeah `/` is a restricted character

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx [hunk 6]
```diff
@@ -30,28 +34,46 @@ export type usePipelineResponse = {
 export const usePipeline = (
   ppsContext: PpsContext | undefined,
   settings: MountSettings,
-  saveNotebookMetaData: (metadata: SameMetadata) => void,
+  saveNotebookMetaData: (metadata: PpsMetadata) => void,
 ): usePipelineResponse => {
   const [loading, setLoading] = useState(false);
-  const [pipelineName, setPipelineName] = useState('');
+  const [pipeline, _setPipeline] = useState({name: ''} as Pipeline);
```
[the remark this answers is not in the document]
Comment (bbonenfant) [reply jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:40]: ok, I'll copy what you did and change this to `setPipelineInternal`

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx (hunk 3 above)
Comment (bbonenfant) [reply jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:101]: This relates to the [this ticket](https://pachyderm.atlassian.net/browse/INT-925). I couldn't figure out how to actually get the message sent from the backend. I've been looking into this but I haven't had much luck yet. Nonetheless I think it should be addressed in future work (and hopefully before release)

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (chainlink) [reply jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:40]: Sorry, not to be pedantic, but I think we should keep the convention of `setThing` for the `useState` methods and rename the other method to something else.

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx (hunk 3 above)
Comment (chainlink) [reply jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:101]: future work note ^

File: jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx (hunk 5 above)
Comment (bbonenfant) [reply jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx:72]: yes it does

**Pull request merged**